ptxboa_streamlit.py

```diff
@@ -157,47 +157,17 @@
     "Deep-dive countries",
     "Optimization",
 ]:
-    # calculate results over different data dimensions:
-    costs_per_region = calculate_results_list(
-        api, parameter_to_change="region", parameter_list=None
-    )
-    costs_per_scenario = calculate_results_list(
-        api,
-        parameter_to_change="scenario",
-        parameter_list=None,
-    )
-    costs_per_res_gen = calculate_results_list(
-        api,
-        parameter_to_change="res_gen",
-        # TODO: here we remove PV tracking manually, this needs to be fixed in data
-        parameter_list=[
-            x
-            for x in api.get_dimension("res_gen").index.to_list()
-            if x != "PV tracking"
-        ],
-    )
-    costs_per_chain = calculate_results_list(
-        api,
-        parameter_to_change="chain",
-        parameter_list=None,
-        override_session_state={"output_unit": "USD/MWh"},
-    )
-
-    if st.session_state["user_changes_df"] is not None:
-        # calculate results over different data dimensions (without user changes):
-        costs_per_region_without_user_changes = calculate_results_list(
-            api,
-            parameter_to_change="region",
-            parameter_list=None,
-            apply_user_data=False,
+    with st.spinner("Please wait. Running optimization model..."):
```

Comment:
I would prefer this message, as it is also shown (briefly) when cached results are loaded instead of really doing an optimization.

```suggestion
    with st.spinner("Please wait. calculating results..."):
```

Reply:
considered in https://github.com/agoenergy/ptx-boa/commit/b4dfc9194960a8c0b27e4c0b8fca7ebeeff5d83b